scipy/signal/fir_filter_design.py

```diff
@@ -632,12 +637,20 @@ def firwin2(numtaps, freq, gain, nfreqs=None, window='hamming', nyq=None,
     if nfreqs is None:
         nfreqs = 1 + 2 ** int(ceil(log(numtaps, 2)))
 
-    # Tweak any repeated values in freq so that interp works.
-    eps = np.finfo(float).eps
-    for k in range(len(freq)):
-        if k < len(freq) - 1 and freq[k] == freq[k + 1]:
-            freq[k] = freq[k] - eps
-            freq[k + 1] = freq[k + 1] + eps
+    if (d == 0).any():
+        # Tweak any repeated values in freq so that interp works.
+        freq = np.array(freq, copy=True)
+        eps = np.finfo(float).eps * nyq
+        for k in range(len(freq)):
```

Comment:
If we change this `for` statement to

```
        for k in range(len(freq) - 1):
```

then the `if` statement can be simplified to

```
            if freq[k] == freq[k + 1]:
```
That part of the code was copied from the old code, so this is a suggestion for another small improvement to the old code and not a fix to code that @krzysztofpioro wrote.  It is not crucial for this PR, but as long as we're cleaning up some of my old mistakes, might as well get this one, too. :smile: